docs/modules/ROOT/pages/explanations/dns_scheme.adoc

```diff
@@ -0,0 +1,121 @@
+= DNS Scheme
+
+An OpenShift 4 cluster requires using a specific DNS scheme which is described on this page. Most of the values are automatically generated and can't be changed.
+
+== User selectable values
+
+`<base domain>`::
+
+  The base domain for the cluster (input by the customer).
+
+== Automatically generated values
+
+`<cluster id>`::
+
+  The Project Syn Lieutenant cluster id of the cluster (auto generated by the Lieutenant API).
+
+`<role>`::
+
+  Machine role. One of _master_, _infra_, _worker_ and _lb_. Other roles possible when required by the customer (special node groups for builds, GPU support etc.).
+
+`<#>`::
+
+  A automatically randomly defined character sequence with four digits.
+
+`<index>`::
+
+  A zero indexed incrementing number.
+
+`<cluster domain>`::
+
+  `<cluster id>.<base domain>`
+
+`<hostname>`::
+
+  The fully qualified domain name of machines belonging to the cluster.
+
+  `<role><#>.<cluster domain>`
+
+`<external api>`::
+
+  OpenShift API URL to be used by clients external to the cluster.
+
+  `api.<cluster domain>`
+
+`<internal api>`::
+
+  OpenShift API URL to be used by clients internal to the cluster.
+
+  `api-int.<cluster domain>`
+
+`<default app domain>`::
+
+  `*.apps.<cluster domain>`
```

Comment:
When our tooling is easily allowing that, then I'm fine to allow customization of that. Is this true with IPI, or is it only possible with UPI? Depending on that we need to decide if this should be a "standard" customization or a "custom" customization.

Reply:
Good point. I'm not sure if or how easy it is in the IPI case. This would need some verification.
In any case I'd propose to make this a "standard" customization since as we already discussed especially the console and apps domain get really unwieldy by default.

Reply:
Before making it a "standard" customization we need to ensure that IPI allows it, otherwise we could get into the situation that we can't use IPI anymore.

Reply:
Doing this with IPI could be tricky. In theory, we could do the same two step process for preparing the configuration (and tweaking things in between). However, DNS records are created automatically and we might not have the time to create the changed DNS record within time.

When changing the default app domain, two things might happen. The installer tries to create the DNS record for the changed domain and fails. The installer might not try to create a DNS record for the altered domain and things start to fail since that record does not exist.

Reply:
We have decided to review this at a later point in time after we have tested this with.